astropy/units/tests/test_quantity_erfa_ufuncs.py

```diff
@@ -528,7 +542,7 @@ def setup_class(self):
     def test_unit_errors(self):
         """Test unit errors when dimensionless parameters are used"""
 
-        msg = "'NoneType' object has no attribute 'get_converter'"
+        msg = "'NoneType'.*no attribute 'get_converter'.*\n.*treated as dimensionless"
```

Comment:
I think this broke "exotic archs" regex. See https://github.com/astropy/astropy/actions/runs/10554512360/job/29236483624

Reply:
```
      def test_unit_errors(self):
          """Test unit errors when dimensionless parameters are used"""
      
          msg = "'NoneType'.*no attribute 'get_converter'.*\n.*treated as dimensionless"
  >       with pytest.raises(AttributeError, match=msg):
  E       AssertionError: Regex pattern did not match.
  E        Regex: "'NoneType'.*no attribute 'get_converter'.*\n.*treated as dimensionless"
  E        Input: "'NoneType' object has no attribute 'get_converter'"
  
  astropy/units/tests/test_quantity_erfa_ufuncs.py:546: AssertionError
```

Reply:
Anyways, I opened https://github.com/astropy/astropy/issues/16884 as follow-up issue that would block v7.0 release. If no one has idea how to fix, my proposal would be to revert the exception behavior changes (but keep your clearer message). Thanks!